### ISSUE TITLE
Support custom permissions

### ISSUE THREAD

**Issue by parkerbxyz** [actions/create-github-app-token#3]
follow up to: https://github.com/gr2m/github-app-token-action/pull/1/commits/2d5eced5968e6735fec0d8906d277ec8c8acc4b3

Our idea is to add separate `permission_*` parameters for each permission supported by GitHub Apps:
https://docs.github.com/en/rest/overview/permissions-required-for-github-apps

The `permission_*` keys can be generated based off https://github.com/octokit/app-permissions. Each time a new version of `@octokit/app-permissions` is released, the `action.yml` could be updated, in order to keep permissions always up-to-date. The `README.md` should be updated as well to make sure the documentation is up-to-date as well. Most permissions can be set to `read` or `write`, some can also be set to `admin`.

Having separate permissions has the benefit of code intelligence and errors shown directly on GitHub when an unknown permission is set due to a typo.

**Comment by gr2m** [comment]
`@octokit/app-permissions` has a new release that I've been waiting for, it fixes a lot of incorrect and missing permissions: https://github.com/octokit/app-permissions/releases/tag/v2.0.0.

**Comment by rvermeulen** [comment]
@parkerbxyz  @gr2m is still being worked on?

**Comment by gr2m** [comment]
yes, we just don't have much time for it and had to prioritize https://github.com/actions/create-github-app-token/pull/143. 

We started to work on a script that auto-updates the `action.yml` file to add all supported permissions as separate input variables, so that you get the full benefit of type checking in your dev environment and on github.com:
https://github.com/actions/create-github-app-token/pull/168
